### DISCUSSION TITLE
How do I install this?

### DISCUSSION THREAD

**Discussion by jatgm1** [dolezsa/thermal_comfort#29] [Q&A] [answered]
This is nowhere to be found in hacs, am I doing something wrong? Thermal_comfort and thermal comfort are unavailable in hacs for me somehow.

**Comment by jatgm1** [comment, marked as answer]
Just realized I can add it by adding the url of the GitHub directly, Totally just guessed based on what I did with another thing. Would probably be kinda cool if this was an instruction on your main page explaining how to do that.

**Reply by rautesamtr** [reply]
Projects that don't fulfill all requirements can't be just added to the hacs default repository (will try to work on that). So until then i hopped that the hacs custom banner is enough since it is described in the [hacs documentation](https://hacs.xyz/docs/faq/custom_repositories).

**Reply by jatgm1** [reply]
If myself or, erm, anyone else I know read the instruction manual I would wholeheartedly agree. But generally I look for a simple instruction for adding it on the creators page because I assume they are the best to inform me on how to install it. Generally I look for a "installation - " thing and like a little paragraph explaining what's up. I actually just explained it on a forum post on home assistant forum for anyone else but I'm not even 100% sure wat I just did. I think I added it under hacs and under add custom repository. But yeah it's fine the way it is if you can figure out what it our doing. I actually just copied your main page url and guessed that maybe pasting it in would work, was gonna try to download the zip and stick it somewhere on the device if it didn't work and see what happens.

**Reply by rautesamtr** [reply]
The project now has hacs default support, so now custom repository needed anymore. I also added some rudimentary installations instructions.

**Comment by jatgm1** [comment]
I messed up something, it works now, but is there a way to take the humidity in one location and convert it to the relative humidity it would be at another temperature? Specifically the temperature of a different location? Im trying to keep the environment for my chinchilla optimal and and using the home assistant to do so, and was trying to use the relative humidity of outside to determine if turning a humidifier or dehumidifier is warranted to prevent them continuously switching off because the humidifier makes it too humid and the dehumidifier makes it too dry, just ping ponging back and forth. Of I can say if the humidity is more than x percent greater outside (after conversion to what it would be at the indoor temperature) then do not Turn on humidifier.

**Reply by rautesamtr** [reply]
You could write a python script with the reverse formula of https://github.com/dolezsa/thermal_comfort/blob/ffe7b248885931d65149bb0f222074226bfc3e60/custom_components/thermal_comfort/sensor.py#L214-L222
e.g.
```python
def computeRelativeHumidity(self, temperature, absHumidity): 
    """https://carnotcycle.wordpress.com/2012/08/04/how-to-convert-relative-humidity-to-absolute-humidity/""" 
    absTemperature = temperature + 273.15;
    relHumidity = absHumidity
    relHumidity *= absTemperature
    relHumidity /= 2.1674; 
    relHumidity /= math.exp((17.67 * temperature) / (243.5 + temperature)); 
    relHumidity /= 6.112; 
    return round(relHumidity, 2) 
```
and call it with a command line sensor

**Reply by jatgm1** [reply]
So How would that be done? And what is that doing?  I am not versed in python. I made it move a motor once and was happy. Also not well versed in meteorology. So absolute Temperature is the Temperature plus 273.15? Is this Celsius? And what is with the mathematical sign before the equals sign? What does that do? Is it doing the absolute humidity times the absolute temperature? Because the relative humidity was defined as the absolute humidity? Then divide by 2.1674 the, erm results of relative humidity variable we just changed? The one that says math.exp something something entirely lost me, then divided by 6.112, and what is the 2 that it's returning? Does it Append a number two to the end of it? Do they just always end with two? Or is it doing the result, rounded times two? Also what's a command line sensor?

**Reply by jatgm1** [reply]
So I made this from what you gave me

- name: entering_humidity
        unit_of_measurement: '%'
        state: >
          {% set temperature = states('sensor.mainroomtemperaturec') | float(0) %}
          {% set relHumidity = states('sensor.outdoorhumidityecowittgw') | float(0) %}
          {% set absHumidity = states('sensor.outdoor_absolutehumidity') | float(0) %}
          {% set absTemperature = temperature + 273.15 %}
          {% set relHumidity = absHumidity %}
          {% set relHumidity = relHumidity * absTemperature %}
          {% set relHumidity = relHumidity / 2.1674 %}
          {% set relHumidity = relHumidity / relHumidity**((17.67 * temperature) / (243.5 + temperature)) %}
          {% set relHumidity = relHumidity / 6.112 %}
          {{ relHumidity | round(9) }}

And it spits out 0.035 or something insane, I think something is up, or maybe home assistant can't do this for some reason 
![Screenshot_20211207-233704](https://user-images.githubusercontent.com/34493130/145149099-d54937b3-3f09-4ea6-810b-d3ce2e853e21.jpg)

Here's all the values, it's under 'entering_humidity'

**Reply by rautesamtr** [reply]
can you try?
```jinja
name: entering_humidity
unit_of_measurement: '%'
state: >
{% set temperature = ((states('sensor.mainroomtemperaturec') | float(0))-32)/1.8 %}
{% set relHumidity = states('sensor.outdoorhumidityecowittgw')  | float(0) %}
{% set absHumidity = states('sensor.outdoor_absolutehumidity') | float(0) %}
{% set absTemperature = temperature + 273.15 %}
{% set relHumidity = absHumidity %}
{% set relHumidity = relHumidity * absTemperature %}
{% set relHumidity = relHumidity / 2.1674 %}
{% set relHumidity = relHumidity / 2.718281828459045235360287471352**((17.67 * temperature) / (243.5 + temperature)) %}
{% set relHumidity = relHumidity / 6.112 %}
{{ relHumidity | round(9) }}
```

**Reply by jatgm1** [reply]
What did you change? It's in Celsius already because I converted it, is that what the line you added did in the beginning? Or is it supposed to be in F

**Reply by jatgm1** [reply]
It output a number over 100 I think it's because it was in Celsius maybe and that thing you added converted it to Celsius.

**Reply by jatgm1** [reply]
I stuck the original one that's in F in there and it seems about right for the relative humidity based on how cold it is but I'm not sure.

**Reply by jatgm1** [reply]
![Screenshot_20211208-093321](https://user-images.githubusercontent.com/34493130/145226311-4adf824a-3d14-40f7-95f1-2c2f836808e4.jpg)

**Reply by rautesamtr** [reply]
Yes i converted Fahrenheit to Celsius in:
```jinja
{% set temperature = ((states('sensor.mainroomtemperaturec') | float(0))-32)/1.8 %}
```
But also correctly used the exponential function in
```jinja
{% set relHumidity = relHumidity / 2.718281828459045235360287471352**((17.67 * temperature) / (243.5 + temperature)) %}
```
`exp(x) = e^x` while `e ≈ 2.718281828459045235360287471352`

the ~21% relative humidity should be correct
